### DISCUSSION TITLE
webdav support

### DISCUSSION THREAD

**Discussion by sdalu** [lostisland/faraday#1533] [Ideas / feature requests]
Support WebDAV verbs, at least in `run_request`, as per discussion in https://github.com/lostisland/faraday/issues/1532

Resources:
* WebDAV Extension RFC: https://datatracker.ietf.org/doc/html/rfc4918

**Comment by iMacTia** [comment]
Thank you for kickstarting this discussion @sdalu, to provide some context, here is my comment from #1532:

> I appreciate you were probably looking to make them first-class citizens like the others, and potentially make them methods like Faraday.copy or connection.proppatch, but I feel like that addition feels unnecessary in the core Faraday library.
> 
> If you'd like to see that happening, then my suggestion would be to instead build an extension gem (faraday-webdav?) that adds these verbs to the list of allowed verbs and also defines the missing methods ❤️ .

Happy to further discuss here how that extension would work and look like.
@sdalu is this something you'd be interested in working on?

**Reply by sdalu** [reply]
Though I fell that's something missing from current faraday implemention, I don't fell I would be the right person for that, as I only needed it for a one-time fix on a caldav

**Comment by cub8** [comment]
Hello, I know this discussion topic is quite old (almost 3 years!), but yesterday I've made extension for Faraday which serves exactly this purpose: https://github.com/cub8/faraday-webdav .

It's new and isn't battle tested yet (although, I've added it to one of my team's application instead of Net:HTTP calls), but I would appreciate checking it out and leaving some feedback 🙌
